Userland/Libraries/LibGfx/PortableImageLoaderCommon.h

```diff
@@ -30,158 +30,146 @@ static constexpr Color adjust_color(u16 max_val, Color color)
     return color;
 }
 
-template<typename TValue>
-static bool read_number(Streamer& streamer, TValue* value)
+static inline ErrorOr<u16> read_number(SeekableStream& stream)
 {
-    u8 byte {};
     StringBuilder sb {};
+    u8 byte {};
 
-    while (streamer.read(byte)) {
+    while ((byte = TRY(stream.template read_value<u8>()))) {
```

Comment:
lhs checked if `byte` could be read. rhs checks if the read byte is `!= 0`. That's different. I think you want an `is_eof()` check here and then the `TRY(read<u8>)` in the loop here too (and elsewhere).